modules/urm/impl/src/main/java/com/innowise/educationalsystem/service/impl/AuthServiceImpl.java

```diff
@@ -47,46 +50,50 @@ public class AuthServiceImpl implements AuthService {
 
     @Override
     public JwtDto loginUser(LoginRequestDto request, Long subscriptionId) {
-        UserResponseDto response = findUserByCredentials(request);
-        authenticateUser(request, response);
-        return jwtMapper.rawTokenToDto(generateToken(response, subscriptionId));
+        User foundUser = findUserByCredentials(request);
+        authenticateUser(request, foundUser);
+        return jwtMapper.rawTokenToDto(generateToken(foundUser, subscriptionId));
     }
 
     @Override
+    @Transactional
     public User signUp(UserSignUpRequestDto userSignUpRequestDto) {
         Invite invite = inviteService.getForSignUp(userSignUpRequestDto.getInviteId());
         checkIfUserAlreadyExists(userSignUpRequestDto);
+        User user = invite.getUser();
+        userMapper.updateUserByDto(userSignUpRequestDto, user); // update user according to dto
+        user.setPassword(passwordEncoder.encode(user.getPassword()));
 
-        User newUser = userMapper.signUpDtoToEntity(userSignUpRequestDto);
-        newUser.setPassword(passwordEncoder.encode(userSignUpRequestDto.getPassword()));
-        newUser.setSubscriptionId(invite.getSubscriptionId());
-        newUser.setEmail(invite.getEmail());
-        newUser.setRoles(invite.getRoles());
-
-        newUser = userRepository.save(newUser);
+        user = userRepository.save(user);
```

Comment:
It is better to savenflush (it will be done automatically under the whod, though still better to make it directly